sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -238,11 +238,17 @@ object TableOutputResolver {
 
     if (reordered.length == expectedCols.length) {
       if (matchedCols.size < inputCols.length) {
-        val extraCols = inputCols.filterNot(col => matchedCols.contains(col.name))
-          .map(col => s"${toSQLId(col.name)}").mkString(", ")
-        throw QueryCompilationErrors.incompatibleDataToTableExtraStructFieldsError(
-          tableName, colPath.quoted, extraCols
-        )
+        if (colPath.isEmpty) {
+          val cannotFindCol = expectedCols.filter(col => !matchedCols.contains(col.name)).head.name
+          throw QueryCompilationErrors.incompatibleDataToTableCannotFindDataError(tableName,
+            cannotFindCol)
+        } else {
+          val extraCols = inputCols.filterNot(col => matchedCols.contains(col.name))
+            .map(col => s"${toSQLId(col.name)}").mkString(", ")
+          throw QueryCompilationErrors.incompatibleDataToTableExtraStructFieldsError(
```

Comment:
so top-level column and nested field have different error classes?

Reply:
extra field? I think it's for missing column/field here?

Reply:
so the behavior is inconsistent between top-level columns and struct fields?

Reply:
I still don't quite understand the code here. If it's for extra columns/fields, why do we throw an error for top-level columns?

Reply:
My question is simple: this is the branch of `if (matchedCols.size < inputCols.length) {` which means extra columns, why do we throw `incompatibleDataToTableCannotFindDataError` for top-level columns which seems to be missing columns?

Reply:
`table: [a, b, c], input [a, c, d]`
Extra column and missing column are orthogonal, they can both occur, or only one of them. Can you elaborate on the error checking logic here?